From 501c0fe1f8f2a89d133f6e00b05c414c39ba1791 Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Tue, 7 Jan 2020 17:08:56 +0100
Subject: [PATCH] PONRTSYS-5630: Allow setting Mac Learning Limit to unlimited

Allow to deactivate the mac learning limit again after it was activated
for a bridge port.
---
 drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.c b/drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.c
index 3f6379dc4bae..0adb929ba5fc 100644
--- a/drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.c
+++ b/drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.c
@@ -23478,6 +23478,9 @@ GSW_return_t GSW_BridgePortConfigSet(void *cdev, GSW_BRIDGE_portConfig_t *param)
 			tbl_prog_brdgeport_ingress.val[4] |= (param->nMacLearningLimit & 0xFF);
 			gswdev->brdgeportconfig_idx[param->nBridgePortId].LearningLimit =
 				param->nMacLearningLimit;
+		} else {
+			tbl_prog_brdgeport_ingress.val[4] |= 0xFF;
+			gswdev->brdgeportconfig_idx[param->nBridgePortId].LearningLimit = 0xFF;
 		}
 	}
 
